src/Common/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -55,12 +55,12 @@ internal static SafeFileHandle Open(string path, Interop.libc.OpenFlags flags, i
                 if (Interop.libcoreclr.GetFileInformationFromFd(fd, out buf) != 0)
                 {
                     handle.Dispose();
-                    throw Interop.GetExceptionForIoErrno(Marshal.GetLastWin32Error(), path);
+                    throw Interop.GetExceptionForIoErrno(Interop.System.GetLastErrorInfo(), path);
                 }
                 if ((buf.mode & Interop.libcoreclr.FileTypes.S_IFMT) == Interop.libcoreclr.FileTypes.S_IFDIR)
                 {
                     handle.Dispose();
-                    throw Interop.GetExceptionForIoErrno(Interop.Errors.EACCES, path, isDirectory: true);
+                    throw Interop.GetExceptionForIoErrno(new Interop.ErrorInfo(Interop.Error.EACCES), path, isDirectory: true);
```

Comment:
On second thought, I'll add ErrorInfo.EACCES and the others we use. Explicit conversion doesn't save many characters and implicit could lead to unintended things like calling GetLastError and converting it to ErrorInfo instead of GetLastErrorInfo.